### PR TITLE
Implement click-to-zoom option for images

### PR DESCRIPTION
**Pull request by dipesh-rawat**
Implemented the functionality to enable click-to-zoom for images on the website. This feature enhances user experience by allowing them to view images in a full-screen overlay upon clicking. Core change as part of this pull request introduces a new parameter (`clicktozoom`) in the `figure` shortcode to selectively activate the zoom functionality.

Fixes https://github.com/kubernetes/website/issues/38051

#### Summary of changes

- **custom.scss**: Added CSS rules for full-screen display of `<figure>` elements.

- **js/zoom.js**: Implemented the script to handle image display in an overlay upon click interaction.

- **figure.html**:  The original source code of Docsy's `figure` shortcode is included to create an override with addition of a new parameter called 'clicktozoom', allowing for the selective activation of the zoom feature.

- **head.html**: Added conditional inclusion for `zoom.js`

- **concepts/overview/components.md**: Demonstrated the click-to-zoom feature on this page to showcase its functionality with the new shortcode parameter.

### [Preview Page (click the image)](https://deploy-preview-46260--kubernetes-io-main-staging.netlify.app/docs/concepts/overview/components/)

/area web-development

### CODE REVIEW

File: layouts/shortcodes/figure.html
Comment (dipesh-rawat) [inline layouts/shortcodes/figure.html:1]: The majority of the code in this file comprises of the existing Docsy figure shortcode [(reference here)](https://github.com/google/docsy/blob/main/layouts/shortcodes/figure.html), which the website currently utilizes. This update adds the HTML source code in the repo to introduce a new parameter (clicktozoom)  to enable the click-to-zoom functionality.

**Comment by salaxander** [comment]
Change seems to work as intended in the deploy preview, and I think the functionality is great!

/lgtm

**Comment by sftim** [comment]
(nit) It's hard to see how to unzoom the image; on my 2160p screen, the close icon is small and almost hidden in the top right corner.

/lgtm

**Comment by dipesh-rawat** [comment]
>  the close icon is small and almost hidden in the top right corner.

Addressed this in the latest commit push.

File: assets/scss/_custom.scss [hunk 1]
```diff
@@ -1352,3 +1352,40 @@ div.alert > em.javascript-required {
...
+}
+
+/* Define styles for close button */
+.figure-close-button {
+  position: absolute;
+  top: 20px;
+  right: 20px;
+  cursor: pointer;
+  font-size: 60px;
+  color: #333;
```
Comment (sftim) [inline assets/scss/_custom.scss:1390]: (nit)
Could use CSS `calc()` for responsive sizing here.

Comment (dipesh-rawat) [reply assets/scss/_custom.scss]: Attempted to incorporate this feedback comment in the latest push.

**Pull request merged**